app/questionnaire/questionnaire_schema.py

```diff
@@ -217,17 +241,44 @@ def _get_answer_dependencies(self):
             if 'max_value' in answer and 'answer_id' in answer['max_value']:
                 answer_id = answer['max_value']['answer_id']
                 dependencies[answer_id].append(dependency_id)
+        return dependencies
 
+    def _get_question_level_dependencies(self):
         # Question level dependencies
+        dependencies = defaultdict(list)
         for question in self.questions:
             for calculation in question.get('calculations', []):
                 answer_id = calculation.get('answer_id')
                 if answer_id:
                     for dependency_id in calculation['answers_to_calculate']:
                         dependencies[answer_id].append(dependency_id)
+        return dependencies
 
+    def _get_titles_dependencies_for_all_questions(self):
+        """ evaluate a questions dependencies via when clauses on titles """
+        dependencies = defaultdict(list)
+        for question in self.questions:
+            self._get_titles_dependencies_for_question(dependencies, question)
         return dependencies
 
+    def _get_titles_dependencies_for_question(self, dependencies, question):
```

Comment:
I'm not a fan of mutating parameters. We'd have to test if we can safely merge the dictionary though if keys already exist. We may already have an issue here.